--- qucs-0.0.15/qucs-core/src/check_touchstone.cpp~	2008-04-23 21:00:45.000000000 +0200
+++ qucs-0.0.15/qucs-core/src/check_touchstone.cpp	2009-10-17 16:22:45.423190968 +0200
@@ -101,7 +101,7 @@
     even = 1;
   }
   /* first line determines the number of expected ports */
-  touchstone_options.ports = (int) sqrt ((size - 1) / 2.0);
+  touchstone_options.ports = (int) sqrt ((size - 1.0) / 2.0);
 
   /* check first frequency value */
   if (f < 0.0) {
